File: src/main/java/com/bindglam/utility/compatibility/ItemsAdderCompatibility.java
package com.bindglam.utility.compatibility;

import dev.lone.itemsadder.api.CustomStack;
import dev.lone.itemsadder.api.FontImages.FontImageWrapper;
import net.kyori.adventure.text.Component;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class ItemsAdderCompatibility implements Compatibility {
    @Override
    public @NotNull Component getGlyph(String id, int offsetX) {
        return Component.text(FontImageWrapper.instance(id).setOffset(offsetX).getString());
    }

    @Override
    public @Nullable Component getGlyphOrNull(String id, int offsetX) {
        if(!FontImageWrapper.getNamespacedIdsInRegistry().contains(id))
            return null;
        return getGlyph(id, offsetX);
    }

    @Override
    public @NotNull ItemStack getCustomItem(String id) {
        return CustomStack.getInstance(id).getItemStack();
    }
}

File: core/src/main/kotlin/com/bindglam/utility/compatibility/ItemsAdderCompatibility.kt
package com.bindglam.utility.compatibility

import dev.lone.itemsadder.api.CustomStack
import dev.lone.itemsadder.api.FontImages.FontImageWrapper
import net.kyori.adventure.text.Component
import org.bukkit.inventory.ItemStack

class ItemsAdderCompatibility : Compatibility {
    override fun getGlyphOrNull(id: String, offsetX: Int): Component? {
        if (!FontImageWrapper.getNamespacedIdsInRegistry().contains(id))
            return null

        return Component.text(FontImageWrapper.instance(id).setOffset(offsetX).string)
    }

    override fun getGlyphWidth(id: String): Int {
        if (!FontImageWrapper.getNamespacedIdsInRegistry().contains(id))
            return 0

        return FontImageWrapper.instance(id).width
    }

    override fun getCustomItemOrNull(id: String): ItemStack? {
        if (!CustomStack.isInRegistry(id))
            return null

        return CustomStack.getInstance(id)!!.itemStack
    }

    override fun getCustomItemIdByItemStack(itemStack: ItemStack?): String? {
        return CustomStack.byItemStack(itemStack)?.namespacedID
    }
}